skill-tree /src/main/java/com/RDS/skilltree/Filters/JWTAuthenticationFilter.java

```diff
@@ -0,0 +1,80 @@
+package com.RDS.skilltree.Filters;
+
+import com.RDS.skilltree.Authentication.UserAuthenticationToken;
+import com.RDS.skilltree.User.*;
+import com.RDS.skilltree.utils.FetchAPI;
+import com.RDS.skilltree.utils.JWTUtils;
+import com.RDS.skilltree.utils.RDSUser.Response;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.concurrent.CompletableFuture;
+
+public class JWTAuthenticationFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private JWTUtils jwtUtils;
+
+    @Autowired
+    private FetchAPI fetchAPI;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private UserRepository userRepository;
+
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
```

Comment:
> Use a thread pool. If you are making multiple concurrent requests to the getRDSUserData() method, you should use a thread pool to manage the threads that are used to make those requests. This will help to improve the performance of your application by preventing the creation of too many threads.

Hi @iamitprakash Spring internally uses a thread pool, can we discuss this once

Reply:
@vikhyat187  can you please help me official documentation link

Reply:
I was telling since the user requesting will be one, so one profile will be getting fetched from the RDS backend, should we add a thread here? @iamitprakash 
And that request would have been already having a thread per request.

Reply:
We discussed this offline, in this case each request will have one thread and we are fetching only one user, so we may not require multiple threads in this case.

Reply:
try removing the completable future